src/middlewared/middlewared/plugins/smb_/registry.py

```diff
@@ -88,6 +96,20 @@ class Config:
     async def reg_delparm(self, share, parm):
         return await self.netconf(action='delparm', share=share, args=[parm])
 
+    @private
+    async def reg_getparm(self, share, parm):
+        to_list = ['vfs objects', 'hosts allow', 'hosts deny']
+        try:
+            ret = await self.netconf(action='getparm', share=share, args=[parm])
+        except Exception as e:
+            if f"Error: given parameter '{parm}' is not set." in e.errmsg:
+                # Copy behavior of samba python binding
+                return None
+            else:
+                raise e
```

Comment:
I think this should be just `raise` or it will reraise the exception in na improper way